omx/gstomxaudiodec.c

```diff
@@ -867,29 +862,24 @@ gst_omx_audio_dec_set_format (GstAudioDecoder * decoder, GstCaps * caps)
     GST_DEBUG_OBJECT (self, "Decoder drained and disabled");
   }
 
+  /* Call subclass set_format for input port configuration */
   if (klass->set_format) {
     if (!klass->set_format (self, self->dec_in_port, caps)) {
-      GST_ERROR_OBJECT (self, "Subclass failed to set the new format");
+      GST_ERROR_OBJECT (self, "Subclass failed to set the new input format");
       return FALSE;
     }
   }
 
-  GST_DEBUG_OBJECT (self, "Updating outport port definition");
-  if (gst_omx_port_update_port_definition (self->dec_out_port,
-          NULL) != OMX_ErrorNone)
-    return FALSE;
+  /* Call subclass set_format for output port configuration */
+  if (klass->set_format) {
+    if (!klass->set_format (self, self->dec_out_port, NULL)) {
+      GST_ERROR_OBJECT (self, "Subclass failed to set the new output format");
+      return FALSE;
+    }
+  }
 
-  /* Get codec data from caps */
+  /* Disable codec data object usage */
   gst_buffer_replace (&self->codec_data, NULL);
-  s = gst_caps_get_structure (caps, 0);
-  codec_data = gst_structure_get_value (s, "codec_data");
-  if (codec_data) {
```

Comment:
With this change 's' and 'codec_data' variables become unused, so please remove them in lines 792 and 793.

Reply:
66cee90a7d9904dfc8155591da324492d4f520ef